app/routes/statement.js

```diff
@@ -1,31 +1,34 @@
-const { getFileStream } = require('../storage')
-const joi = require('joi')
-const boom = require('@hapi/boom')
+const Boom = require('@hapi/boom')
+
+const schema = require('./schemas/statement')
+
+const getReadThroughStatement = require('../statement')
 
 module.exports = {
   method: 'GET',
   path: '/statement/{version}/{filename}',
+  options: {
+    validate: {
+      params: schema,
+      failAction: async (request, h, error) => {
+        return Boom.notFound(error)
+      }
+    }
+  },
   handler: async (request, h) => {
     const filename = request.params.filename
 
     try {
-      const statementFile = await getFileStream(filename)
-      return h.response(statementFile.readableStreamBody)
+      const statement = await getReadThroughStatement(request, request.params.filename)
+
+      return h.response(statement)
         .type('application/pdf')
         .header('Connection', 'keep-alive')
         .header('Cache-Control', 'no-cache')
         .header('Content-Disposition', `attachment;filename=${filename}`)
         .code(200)
     } catch (err) {
-      return boom.badRequest(err)
-    }
-  },
-  options: {
-    validate: {
-      params: joi.object({
-        version: joi.string(),
-        filename: joi.string().required()
-      })
+      return Boom.badRequest(err)

```

Comment:
Would this one not be better as not found as at this point we're happy with the request but for whatever reason we've failed to return a statement.  I guess this could be either because the statement doesn't exist or our service broke, so maybe if statement from `getReadThroughStatement` is undefined then we return 404 if an exception is thrown we return 500?

Reply:
Cannot _really_ check for this.

If Blob isn't available, it will error and hit this.

If Blob is available and our service breaks it will be because `Hapi` can't render what we've returned, error and hit this.